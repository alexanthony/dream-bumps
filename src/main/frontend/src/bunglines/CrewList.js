import React, {PropTypes} from 'react';
import { connect } from 'react-redux';
import {Card, CardTitle, CardText} from 'material-ui/Card';

import { fetchCrews } from '../crews/actions';
import { getNumberOfDivisions, getCrewsByDivision, getCurrentDay } from '../rootReducer';
import DivisionTable from './DivisionTable';

class CrewList extends React.Component {
  componentWillMount() {
    this.props.fetchCrews(this.props.sex);
  }

  render() {
    const title = this.props.day >= 4 ? 'Finishing Order' : `Starting Order for Day ${this.props.day + 1}`
    return (
      <Card>
        <CardTitle title={title} />
        <CardText>
          {
            Object.keys(this.props.crews).map(key =>
              (
                <DivisionTable
                  sex={this.props.sex}
                  division={key}
                  key={key}
                  crews={this.props.crews[key]}
                />
              )
            )
          }
        </CardText>
      </Card>
    );
  }
}

CrewList.propTypes = {
  fetchCrews: PropTypes.func,
  sex: PropTypes.string,
  numberOfDivisions: PropTypes.number,
  crews: PropTypes.array,
  day: PropTypes.number
};

const mapStateToProps = (state, ownProps) => ({
  crews: getCrewsByDivision(state)(ownProps.sex),
  numberOfDivisions: getNumberOfDivisions(state)(ownProps.sex),
  day: getCurrentDay(state)
});

export default connect(mapStateToProps, { fetchCrews })(CrewList);

/*
<DivisionTable
  sex={this.props.sex}
  division={key}
  key={key}
  crews={this.props.crews[key]}
/>
*/
