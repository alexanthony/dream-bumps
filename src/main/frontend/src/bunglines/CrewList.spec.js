import React from 'react';
import { shallow } from 'enzyme';
import { CrewList } from './CrewList';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const fetchCrews = jest.fn();
  const component = shallow(
    <CrewList
      fetchCrews={fetchCrews}
      crews={{0: [{}, {}], 1: [{}, {}]}}
      day={3}
      marketOpen
      numberOfDivisions={7}
      sex="male"
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
  expect(fetchCrews).toHaveBeenCalled();
});
