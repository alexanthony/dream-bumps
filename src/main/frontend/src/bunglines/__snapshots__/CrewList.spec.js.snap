exports[`test Renders properly 1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}>
  <CardTitle
    title="Starting Order for Day 4" />
  <CardText>
    <Connect(DivisionTable)
      crews={
        Array [
          Object {},
          Object {},
        ]
      }
      day={3}
      division="0"
      sex="male" />
    <Connect(DivisionTable)
      crews={
        Array [
          Object {},
          Object {},
        ]
      }
      day={3}
      division="1"
      sex="male" />
  </CardText>
</Card>
`;
